app/models/agents/google_translation_agent.rb

```diff
@@ -0,0 +1,56 @@
+require 'google/apis/translate_v2'
+
+module Agents
+  class GoogleTranslationAgent < Agent
+    cannot_be_scheduled!
+
+    description <<-MD
+      The Translation Agent will attempt to translate text between natural languages.
+
+      Services are provided using Google Translate. You can [sign up](https://cloud.google.com/translate/) to get `GOOGLE_API_KEY` which is required to use this agent.
+
+    MD
+
+    event_description "User defined"
+
+    def default_options
+      {
+        'to' => "sv",
+        'from' => 'en',
+        'expected_receive_period_in_days' => 1,
+        'content' => {
+          'text' => "{{message.text}}",
+          'content' => "{{xyz}}"
+        }
+      }
+    end
+
+    def working?
+      last_receive_at && last_receive_at > interpolated['expected_receive_period_in_days'].to_i.days.ago && !recent_error_logs?
+    end
+
+    def validate_options
+      unless ENV['GOOGLE_API_KEY'].present? && options['to'].present? && options['content'].present? && options['expected_receive_period_in_days'].present?
```

Comment:
You know I had the same thought when writing the spec and revisiting the `TranslatorAgent`, I guess I was just stuck in work routine always using ENV.